src/main/java/com/odysseusinc/arachne/portal/service/impl/BaseDataSourceServiceImpl.java

```diff
@@ -267,11 +266,38 @@ public DS findByUuidUnsecured(String uuid) throws NotExistException {
     public void delete(Long id) throws IOException, SolrServerException {
 
         log.info("Deleting datasource with id={}", id);
-        if (dataSourceRepository.deleteByIdAndDeletedIsNull(id) == 0) {
+        dataSourceRepository.delete(id);
+ /*       if (dataSourceRepository.deleteByIdAndDeletedIsNull(id) == 0) {//
             throw new NotExistException(getType());
+        }*/
+    }
+
+    @PreAuthorize("hasPermission(#id, 'DataSource', "
+            + "T(com.odysseusinc.arachne.portal.security.ArachnePermission).EDIT_DATASOURCE)")
+    @Transactional
+    @Override
+    public void unpublish(Long id) throws IOException, SolrServerException {
+
+        log.info("Unpublishing datasource with id={}", id);
+        DS dataSource = getByIdUnsecured(id);
+
+        if (dataSource.getPublished()){
+            dataSource.setPublished(false);
+            dataSourceRepository.save(dataSource);
+
+            solrService.deleteByQuery(SolrServiceImpl.DATA_SOURCE_COLLECTION, "id:" + id);
```

Comment:
Solr throws checked expression, therefore transaction will not be rolled back, when solr fails
(https://docs.spring.io/spring/docs/2.5.x/reference/transaction.html#transaction-declarative)